### ISSUE TITLE
Missing Pages/Errors on Showcase Demo

### ISSUE THREAD

**Issue by tmulle** [melloware/quarkus-faces#165]
I just clicked on every link in the demo that referred to ‘localhost’ and I found a few pages that gave a 404/error when running in the Docker image.

[Expansion](http://localhost:8080/ui/data/datatable/expansion.xhtml?jfwid=e58fa) - 404/error
The error in the console is:
```
The Datatable expansion fails with:
2022-08-17 16:40:27,019 SEVERE [org.pri.app.exc.PrimeExceptionHandler] (executor-thread-31) DataTable#rowKey must be defined for component j_id_7i:j_id_7j: java.lang.UnsupportedOperationException: DataTable#rowKey must be defined for component j_id_7i:j_id_7j
	at org.primefaces.component.datatable.DataTable.getRowKey(DataTable.java:635)
	at org.primefaces.component.datatable.DataTableRenderer.encodeRow(DataTableRenderer.java:1237)
	at org.primefaces.component.datatable.DataTableRenderer.encodeRows(DataTableRenderer.java:1180)
	at org.primefaces.component.datatable.DataTableRenderer.encodeTbody(DataTableRenderer.java:1120)
	at org.primefaces.component.datatable.DataTableRenderer.encodeTbody(DataTableRenderer.java:1073)
	at org.primefaces.component.datatable.DataTableRenderer.encodeRegularTable(DataTableRenderer.java:402)
	at org.primefaces.component.datatable.DataTableRenderer.encodeMarkup(DataTableRenderer.java:332)
	at org.primefaces.component.datatable.DataTableRenderer.render(DataTableRenderer.java:103)
	at org.primefaces.component.datatable.DataTableRenderer.encodeEnd(DataTableRenderer.java:94)
	at javax.faces.component.UIComponentBase.encodeEnd(UIComponentBase.java:649)
	at javax.faces.component.UIData.encodeEnd(UIData.java:1649)
	at javax.faces.component.UIComponentBase.encodeAll(UIComponentBase.java:530)
	at javax.faces.component.UIComponentBase.encodeAll(UIComponentBase.java:526)
	at javax.faces.component.UIComponentBase.encodeAll(UIComponentBase.java:526)
	at javax.faces.component.UIComponentBase.encodeAll(UIComponentBase.java:526)
	at org.apache.myfaces.view.facelets.FaceletViewDeclarationLanguage.renderView(FaceletViewDeclarationLanguage.java:1782)
	at org.apache.myfaces.application.ViewHandlerImpl.renderView(ViewHandlerImpl.java:316)
	at javax.faces.application.ViewHandlerWrapper.renderView(ViewHandlerWrapper.java:74)
```

[MegaMenu](http://localhost:8080/ui/menu/megaMenu.xhtml?jfwid=e58fa) - 404/error
Error in console:
```
2022-08-17 17:16:38,043 SEVERE [org.pri.app.exc.PrimeExceptionHandler] (executor-thread-45) class org.apache.myfaces.view.facelets.compiler.UIInstructions cannot be cast to class org.primefaces.model.menu.MenuElement (org.apache.myfaces.view.facelets.compiler.UIInstructions and org.primefaces.model.menu.MenuElement are in unnamed module of loader io.quarkus.bootstrap.runner.RunnerClassLoader @7cd62f43): java.lang.ClassCastException: class org.apache.myfaces.view.facelets.compiler.UIInstructions cannot be cast to class org.primefaces.model.menu.MenuElement (org.apache.myfaces.view.facelets.compiler.UIInstructions and org.primefaces.model.menu.MenuElement are in unnamed module of loader io.quarkus.bootstrap.runner.RunnerClassLoader @7cd62f43)
	at org.primefaces.component.megamenu.MegaMenuRenderer.encodeColumn(MegaMenuRenderer.java:213)
	at org.primefaces.component.megamenu.MegaMenuRenderer.encodeRootSubmenu(MegaMenuRenderer.java:186)
	at org.primefaces.component.megamenu.MegaMenuRenderer.encodeRootItems(MegaMenuRenderer.java:116)
	at org.primefaces.component.megamenu.MegaMenuRenderer.encodeMarkup(MegaMenuRenderer.java:85)
	at org.primefaces.component.menu.BaseMenuRenderer.encodeEnd(BaseMenuRenderer.java:51)
	at javax.faces.component.UIComponentBase.encodeEnd(UIComponentBase.java:649)
	at javax.faces.component.UIComponentBase.encodeAll(UIComponentBase.java:530)
	at javax.faces.component.UIComponentBase.encodeAll(UIComponentBase.java:526)
	at javax.faces.component.UIComponentBase.encodeAll(UIComponentBase.java:526)
	at org.apache.myfaces.view.facelets.FaceletViewDeclarationLanguage.renderView(FaceletViewDeclarationLanguage.java:1782)
	at org.apache.myfaces.application.ViewHandlerImpl.renderView(ViewHandlerImpl.java:316)
	at javax.faces.application.ViewHandlerWrapper.renderView(ViewHandlerWrapper.java:74)
	at org.omnifaces.viewhandler.OmniViewHandler.renderView(OmniViewHandler.java:151)
	at javax.faces.application.ViewHandlerWrapper.renderView(ViewHandlerWrapper.java:74)
	at org.apache.myfaces.lifecycle.RenderResponseExecutor.execute(RenderResponseExecutor.java:122)
	at org.apache.myfaces.lifecycle.LifecycleImpl.render(LifecycleImpl.java:241)
	at javax.faces.lifecycle.LifecycleWrapper.render(LifecycleWrapper.java:43)
	at javax.faces.webapp.FacesServlet.service(FacesServlet.java:209)
	at io.undertow.servlet.handlers.ServletHandler.handleRequest(ServletHandler.java:74)
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:129)
	at org.omnifaces.filter.CacheControlFilter.doFilter(CacheControlFilter.java:239)
	at org.omnifaces.filter.HttpFilter.doFilter(HttpFilter.java:108)
	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
	at org.omnifaces.filter.GzipResponseFilter.doFilter(GzipResponseFilter.java:183)
	at org.omnifaces.filter.HttpFilter.doFilter(HttpFilter.java:108)
	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
	at org.omnifaces.filter.CharacterEncodingFilter.doFilter(CharacterEncodingFilter.java:123)
	at org.omnifaces.filter.HttpFilter.doFilter(HttpFilter.java:108)
	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
	at io.undertow.servlet.handlers.FilterHandler.handleRequest(FilterHandler.java:84)
	at io.undertow.servlet.handlers.security.ServletSecurityRoleHandler.handleRequest(ServletSecurityRoleHandler.java:63)
	at io.undertow.servlet.handlers.ServletChain$1.handleRequest(ServletChain.java:68)
	at io.undertow.servlet.handlers.ServletDispatchingHandler.handleRequest(ServletDispatchingHandler.java:36)
	at io.undertow.servlet.handlers.RedirectDirHandler.handleRequest(RedirectDirHandler.java:67)
	at io.undertow.servlet.handlers.security.SSLInformationAssociationHandler.handleRequest(SSLInformationAssociationHandler.java:133)
	at io.undertow.servlet.handlers.security.ServletAuthenticationCallHandler.handleRequest(ServletAuthenticationCallHandler.java:57)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.undertow.security.handlers.AbstractConfidentialityHandler.handleRequest(AbstractConfidentialityHandler.java:46)
	at io.undertow.servlet.handlers.security.ServletConfidentialityConstraintHandler.handleRequest(ServletConfidentialityConstraintHandler.java:65)
	at io.undertow.security.handlers.AuthenticationMechanismsHandler.handleRequest(AuthenticationMechanismsHandler.java:60)
	at io.undertow.servlet.handlers.security.CachedAuthenticatedSessionHandler.handleRequest(CachedAuthenticatedSessionHandler.java:77)
	at io.undertow.security.handlers.NotificationReceiverHandler.handleRequest(NotificationReceiverHandler.java:50)
	at io.undertow.security.handlers.AbstractSecurityContextAssociationHandler.handleRequest(AbstractSecurityContextAssociationHandler.java:43)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.quarkus.undertow.runtime.RewriteCorrectingHandlerWrappers$PostWrapper$1.handleRequest(RewriteCorrectingHandlerWrappers.java:51)
	at io.undertow.predicate.PredicatesHandler.handleRequest(PredicatesHandler.java:110)
	at io.undertow.server.handlers.SecureCookieHandler.handleRequest(SecureCookieHandler.java:54)
	at io.undertow.predicate.PredicatesHandler.handleRequest(PredicatesHandler.java:93)
	at io.quarkus.undertow.runtime.RewriteCorrectingHandlerWrappers$PreWrapper$1.handleRequest(RewriteCorrectingHandlerWrappers.java:31)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.undertow.servlet.handlers.ServletInitialHandler.handleFirstRequest(ServletInitialHandler.java:247)
	at io.undertow.servlet.handlers.ServletInitialHandler.access$100(ServletInitialHandler.java:56)
	at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:111)
	at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:108)
	at io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:48)
	at io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43)
	at io.quarkus.undertow.runtime.UndertowDeploymentRecorder$9$1.call(UndertowDeploymentRecorder.java:595)
	at io.undertow.servlet.handlers.ServletInitialHandler.dispatchRequest(ServletInitialHandler.java:227)
	at io.undertow.servlet.handlers.ServletInitialHandler.handleRequest(ServletInitialHandler.java:152)
	at io.quarkus.undertow.runtime.UndertowDeploymentRecorder$1.handleRequest(UndertowDeploymentRecorder.java:120)
	at io.undertow.server.Connectors.executeRootHandler(Connectors.java:284)
	at io.undertow.server.DefaultExchangeHandler.handle(DefaultExchangeHandler.java:18)
	at io.quarkus.undertow.runtime.UndertowDeploymentRecorder$5$1.run(UndertowDeploymentRecorder.java:417)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:555)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

[Autocomplete](http://localhost:8080/ui/misc/terminal/autocomplete.xhtml?jfwid=e58fa) - Loads but autocomplete doesn’t work in Safari.. I try the samples like ‘git + TAB’ and nothing happens (no errors in console)

PDF exporter on DataExporter -> [Customized](http://localhost:8080/ui/data/dataexporter/customizedDocuments.xhtml?jfwid=e58fa) (loads but failed when I clicked the PDF button, XLS works)

Error:
```
2022-08-17 16:36:03,342 SEVERE [org.pri.app.exc.PrimeExceptionHandler] (executor-thread-29) /deployments/null/resources/demo/images/prime_logo.png (No such file or directory): java.io.FileNotFoundException: /deployments/null/resources/demo/images/prime_logo.png (No such file or directory)
	at java.base/java.io.FileInputStream.open0(Native Method)
	at java.base/java.io.FileInputStream.open(FileInputStream.java:219)
	at java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)
	at java.base/java.io.FileInputStream.<init>(FileInputStream.java:112)
	at java.base/sun.net.www.protocol.file.FileURLConnection.connect(FileURLConnection.java:86)
	at java.base/sun.net.www.protocol.file.FileURLConnection.getInputStream(FileURLConnection.java:184)
	at java.base/java.net.URL.openStream(URL.java:1165)
	at com.lowagie.text.Image.getInstance(Image.java:252)
	at com.lowagie.text.Image.getInstance(Image.java:328)
	at org.primefaces.showcase.view.data.dataexporter.CustomizedDocumentsView.preProcessPDF(CustomizedDocumentsView.java:141)
	at org.primefaces.showcase.view.data.dataexporter.CustomizedDocumentsView_ClientProxy.preProcessPDF(Unknown Source)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.apache.el.parser.AstValue.invoke(AstValue.java:247)
	at org.apache.el.MethodExpressionImpl.invoke(MethodExpressionImpl.java:267)
	at org.apache.myfaces.view.facelets.el.ContextAwareTagMethodExpression.invoke(ContextAwareTagMethodExpression.java:96)
	at org.primefaces.component.datatable.export.DataTablePDFExporter.preExport(DataTablePDFExporter.java:85)
	at org.primefaces.component.datatable.export.DataTableExporter.export(DataTableExporter.java:206)
	at org.primefaces.component.export.DataExporter.processAction(DataExporter.java:171)
	at javax.faces.event.ActionEvent.processListener(ActionEvent.java:58)
	at javax.faces.component.UIComponentBase.broadcast(UIComponentBase.java:403)
	at javax.faces.component.UICommand.broadcast(UICommand.java:55)
	at javax.faces.component.UIViewRoot._broadcastAll(UIViewRoot.java:1238)
	at javax.faces.component.UIViewRoot.broadcastEvents(UIViewRoot.java:414)
	at javax.faces.component.UIViewRoot._process(UIViewRoot.java:1718)
	at javax.faces.component.UIViewRoot.processApplication(UIViewRoot.java:920)
	at org.apache.myfaces.lifecycle.InvokeApplicationExecutor.execute(InvokeApplicationExecutor.java:43)
	at org.apache.myfaces.lifecycle.LifecycleImpl.executePhase(LifecycleImpl.java:172)
	at org.apache.myfaces.lifecycle.LifecycleImpl.execute(LifecycleImpl.java:125)
	at javax.faces.lifecycle.LifecycleWrapper.execute(LifecycleWrapper.java:61)
	at javax.faces.webapp.FacesServlet.service(FacesServlet.java:207)
	at io.undertow.servlet.handlers.ServletHandler.handleRequest(ServletHandler.java:74)
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:129)
	at org.omnifaces.filter.CacheControlFilter.doFilter(CacheControlFilter.java:239)
	at org.omnifaces.filter.HttpFilter.doFilter(HttpFilter.java:108)
	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
	at org.omnifaces.filter.GzipResponseFilter.doFilter(GzipResponseFilter.java:183)
	at org.omnifaces.filter.HttpFilter.doFilter(HttpFilter.java:108)
	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
	at org.omnifaces.filter.CharacterEncodingFilter.doFilter(CharacterEncodingFilter.java:123)
	at org.omnifaces.filter.HttpFilter.doFilter(HttpFilter.java:108)
	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
	at io.undertow.servlet.handlers.FilterHandler.handleRequest(FilterHandler.java:84)
	at io.undertow.servlet.handlers.security.ServletSecurityRoleHandler.handleRequest(ServletSecurityRoleHandler.java:63)
	at io.undertow.servlet.handlers.ServletChain$1.handleRequest(ServletChain.java:68)
	at io.undertow.servlet.handlers.ServletDispatchingHandler.handleRequest(ServletDispatchingHandler.java:36)
	at io.undertow.servlet.handlers.RedirectDirHandler.handleRequest(RedirectDirHandler.java:67)
	at io.undertow.servlet.handlers.security.SSLInformationAssociationHandler.handleRequest(SSLInformationAssociationHandler.java:133)
	at io.undertow.servlet.handlers.security.ServletAuthenticationCallHandler.handleRequest(ServletAuthenticationCallHandler.java:57)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.undertow.security.handlers.AbstractConfidentialityHandler.handleRequest(AbstractConfidentialityHandler.java:46)
	at io.undertow.servlet.handlers.security.ServletConfidentialityConstraintHandler.handleRequest(ServletConfidentialityConstraintHandler.java:65)
	at io.undertow.security.handlers.AuthenticationMechanismsHandler.handleRequest(AuthenticationMechanismsHandler.java:60)
	at io.undertow.servlet.handlers.security.CachedAuthenticatedSessionHandler.handleRequest(CachedAuthenticatedSessionHandler.java:77)
	at io.undertow.security.handlers.NotificationReceiverHandler.handleRequest(NotificationReceiverHandler.java:50)
	at io.undertow.security.handlers.AbstractSecurityContextAssociationHandler.handleRequest(AbstractSecurityContextAssociationHandler.java:43)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.quarkus.undertow.runtime.RewriteCorrectingHandlerWrappers$PostWrapper$1.handleRequest(RewriteCorrectingHandlerWrappers.java:51)
	at io.undertow.predicate.PredicatesHandler.handleRequest(PredicatesHandler.java:110)
	at io.undertow.server.handlers.SecureCookieHandler.handleRequest(SecureCookieHandler.java:54)
	at io.undertow.predicate.PredicatesHandler.handleRequest(PredicatesHandler.java:93)
	at io.quarkus.undertow.runtime.RewriteCorrectingHandlerWrappers$PreWrapper$1.handleRequest(RewriteCorrectingHandlerWrappers.java:31)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.undertow.servlet.handlers.ServletInitialHandler.handleFirstRequest(ServletInitialHandler.java:247)
	at io.undertow.servlet.handlers.ServletInitialHandler.access$100(ServletInitialHandler.java:56)
	at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:111)
	at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:108)
	at io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:48)
	at io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43)
	at io.quarkus.undertow.runtime.UndertowDeploymentRecorder$9$1.call(UndertowDeploymentRecorder.java:595)
	at io.undertow.servlet.handlers.ServletInitialHandler.dispatchRequest(ServletInitialHandler.java:227)
	at io.undertow.servlet.handlers.ServletInitialHandler.handleRequest(ServletInitialHandler.java:152)
	at io.quarkus.undertow.runtime.UndertowDeploymentRecorder$1.handleRequest(UndertowDeploymentRecorder.java:120)
	at io.undertow.server.Connectors.executeRootHandler(Connectors.java:284)
	at io.undertow.server.DefaultExchangeHandler.handle(DefaultExchangeHandler.java:18)
	at io.quarkus.undertow.runtime.UndertowDeploymentRecorder$5$1.run(UndertowDeploymentRecorder.java:417)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:555)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

**Comment by melloware** [comment]
Fixed all issues except the AutoComplete.  I am not seeing any issues with AutoComplete but i may be doing something different.  Does it work for you on PF Showcase: https://www.primefaces.org/showcase/ui/input/autoComplete.xhtml

I am also not using Safari

**Comment by tmulle** [comment]
That should be the TERMINAL example.. the regular autocomplete works fine for text input boxes.

This is the terminal component and the autocomplete feature.

Autocomplete <https://www.primefaces.org/showcase/ui/misc/terminal/autocomplete.xhtml?jfwid=4a595>


No errors in the console but the JS error is `JSON Parse Error: Unrecognized token ‘&’` in core.js.html:16

The live demo works fine on the internet.. just not running locally

**Comment by melloware** [comment]
OK I am seeing this error in the console in the Terminal AutoComplete!

![image](https://user-images.githubusercontent.com/4399574/185205647-d8799e5e-c3a5-4dcd-9d33-662ec567b95b.png)

**Comment by melloware** [comment]
Interesting. PF Showcase on Mojarra:
![image](https://user-images.githubusercontent.com/4399574/185206133-0dc755a0-fb48-48c4-b3c1-36f315bdfe6f.png)

Quarkus running MyFaces:
![image](https://user-images.githubusercontent.com/4399574/185206255-3d5311da-a593-480b-b580-47546766d689.png)


Looks like the response JSON is being encoded.

**Comment by melloware** [comment]
Thanks for all the help!  Reported and fixed the issue in PF 12.0.0-RC3.

**Comment by tmulle** [comment]
No problem! Glad I could help..
